src/lib.rs

```diff
@@ -218,12 +226,11 @@ impl RustafarianDrone {
             Some(channel) => {
                 // Check if packet can be dropped, if so check the PDR
                 if !skip_pdr_check && self.should_drop() {
-                    // Notify controller that a packet has been dropped
-                    self.controller_send.send(DroneEvent::PacketDropped(packet.clone()));
-
                     // Packet dropped
-                    self.send_nack_fragment(packet, NackType::Dropped, fragment_index);
+                    self.send_nack_fragment(packet.clone(), NackType::Dropped, fragment_index);
```

Comment:
Why did you swap the order back? I think the correct way is to first notify the controller, then send a NACK.

Reply:
older commit. I just pushed the saturation

Reply:
Then swap these two lines again, otherwise it's an incorrect behavior